--- ../avorion-stock/data/scripts/events/pirateattack.lua	2019-02-10 13:06:21.603399100 -0600
+++ ../avorion-event-balance/data/scripts/events/pirateattack.lua	2019-02-26 04:53:30.417212100 -0600
@@ -6,6 +6,11 @@
 require ("galaxy")
 require ("randomext")
 require ("stringutility")
+
+-- <dcc title="require event balancer">
+require("dcc-event-balance/main")
+-- </dcc>
+
 local Placer = require("placer")
 local AsyncPirateGenerator = require ("asyncpirategenerator")
 local UpgradeGenerator = require ("upgradegenerator")
@@ -50,6 +55,14 @@
         return
     end
 
+    -- <dcc title="determine if the event should be skipped">
+    if EventBalance.ShouldSkipEvent({script="pirates-being-annoying"}) then
+        print("[EB] pirate event skipped")
+        terminate()
+        return
+    end
+    -- </dcc>
+
     ships = {}
     participants = {}
     reward = 0
@@ -123,7 +136,9 @@
 end
 
 function PirateAttack.getUpdateInterval()
-    return 15
+    -- <dcc title="increase event delay">
+    return 15 * EventBalance.PauseMultiplier
+    -- </dcc>
 end
 
 function PirateAttack.onPiratesGenerated(generated)
