--- ../avorion-stock/data/scripts/player/eventscheduler.lua	2019-02-26 13:08:19.243918700 -0600
+++ ../avorion-event-balance/data/scripts/player/eventscheduler.lua	2019-02-26 13:32:59.843285500 -0600
@@ -5,6 +5,10 @@
 require ("randomext")
 require ("utility")
 
+-- <dcc title="require event balancer">
+require ("dcc-event-balance/main")
+-- </dcc>
+
 local events =
 {
     {schedule = random():getInt(45, 60) * 60, script = "convoidistresssignal", arguments = {true}, to = 560},
@@ -18,7 +22,10 @@
     {schedule = random():getInt(50, 70) * 60, script = "spawntravellingmerchant", to = 520},
 }
 
-local pause = 5 * 60
+-- <dcc title="increase event delay">
+local pause = (5 * 60) * EventBalance.PauseMultiplier
+-- </dcc>
+
 local pauseTime = pause
 
 -- Don't remove or alter the following comment, it tells the game the namespace this script lives in. If you remove it, the script will break.
@@ -58,8 +65,10 @@
         if p.index < player.index then return end
     end
 
-    -- but, if we're not alone, we speed up events by 50%
-    if #players > 1 then timeStep = timeStep * 1.5 end
+    -- <dcc title="disable multiple players speeding up events">
+    -- -- but, if we're not alone, we speed up events by 50%
+    -- if #players > 1 then timeStep = timeStep * 1.5 end
+    -- </dcc>
 
     if pauseTime > 0 then
         pauseTime = pauseTime - timeStep
@@ -77,6 +86,15 @@
 
             local position = length(vec2(Sector():getCoordinates()))
             if position >= from and position <= to then
+
+                -- <dcc title="determine if event should be skipped">
+                if EventBalance.ShouldSkipEvent(event) then
+                    print("[EB] event `" .. event.script .. "` skipped")
+                    event.time = event.schedule
+                    return
+                end
+                -- </dcc>
+
                 -- start event
                 local arguments = event.arguments or {}
                 Player():addScriptOnce(event.script, unpack(arguments))
