--- ../avorion-stock/data/scripts/events/pirateattack.lua	2019-02-26 13:08:19.573710500 -0600
+++ ../avorion-event-balance/data/scripts/events/pirateattack.lua	2019-02-26 13:31:17.209229400 -0600
@@ -7,6 +7,11 @@
 require ("randomext")
 require ("stringutility")
 require("player")
+
+-- <dcc title="require event balancer">
+require("dcc-event-balance/main")
+-- </dcc>
+
 local Placer = require("placer")
 local AsyncPirateGenerator = require ("asyncpirategenerator")
 local UpgradeGenerator = require ("upgradegenerator")
@@ -52,6 +57,14 @@
         return
     end
 
+    -- <dcc title="determine if the event should be skipped">
+    if EventBalance.ShouldSkipEvent({script="pirates-being-annoying"}) then
+        print("[EB] pirate event skipped")
+        terminate()
+        return
+    end
+    -- </dcc>
+
     ships = {}
     participants = {}
     reward = 0
@@ -126,7 +139,9 @@
 end
 
 function PirateAttack.getUpdateInterval()
-    return 15
+    -- <dcc title="increase event delay">
+    return 15 * EventBalance.PauseMultiplier
+    -- </dcc>
 end
 
 function PirateAttack.onPiratesGenerated(generated)
