--- ../avorion-0.11.0.7844/data/scripts/events/pirateattack.lua	2017-03-19 16:55:36.660981300 -0500
+++ ../avorion-event-balance/data/scripts/events/pirateattack.lua	2017-03-28 12:26:34.714666700 -0500
@@ -6,6 +6,11 @@
 require ("galaxy")
 require ("randomext")
 require ("stringutility")
+
+-- <dcc title="require event balancer">
+require("dcc-event-balance/main")
+-- </dcc>
+
 local PirateGenerator = require ("pirategenerator")
 local UpgradeGenerator = require ("upgradegenerator")
 local TurretGenerator = require ("turretgenerator")
@@ -43,6 +48,14 @@
         return
     end
 
+    -- <dcc title="determine if the event should be skipped">
+    if EventBalance.ShouldSkipEvent({script="pirates-being-annoying"}) then
+        print("[EB] pirate event skipped")
+        terminate()
+        return
+    end
+    -- </dcc>
+
     ships = {}
     participants = {}
     reward = 0
@@ -158,7 +171,9 @@
 end
 
 function getUpdateInterval()
-    return 15
+    -- <dcc title="increase event delay">
+    return 15 * EventBalance.PauseMultiplier
+    -- </dcc>
 end
 
 function update(timeStep)
